features/vaadin-surveillance-views/src/main/java/org/opennms/features/vaadin/surveillanceviews/service/DefaultSurveillanceViewService.java

```diff
@@ -0,0 +1,705 @@
+package org.opennms.features.vaadin.surveillanceviews.service;
+
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import org.opennms.core.criteria.Alias;
+import org.opennms.core.criteria.Criteria;
+import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.core.criteria.Order;
+import org.opennms.core.criteria.restrictions.Restriction;
+import org.opennms.core.criteria.restrictions.Restrictions;
+import org.opennms.features.vaadin.surveillanceviews.config.SurveillanceViewProvider;
+import org.opennms.features.vaadin.surveillanceviews.model.Category;
+import org.opennms.features.vaadin.surveillanceviews.model.View;
+import org.opennms.netmgt.config.GroupDao;
+import org.opennms.netmgt.config.groups.Group;
+import org.opennms.netmgt.dao.api.AlarmDao;
+import org.opennms.netmgt.dao.api.CategoryDao;
+import org.opennms.netmgt.dao.api.GraphDao;
+import org.opennms.netmgt.dao.api.MonitoredServiceDao;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.dao.api.NotificationDao;
+import org.opennms.netmgt.dao.api.OutageDao;
+import org.opennms.netmgt.dao.api.ResourceDao;
+import org.opennms.netmgt.model.OnmsAlarm;
+import org.opennms.netmgt.model.OnmsCategory;
+import org.opennms.netmgt.model.OnmsMonitoredService;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.OnmsNotification;
+import org.opennms.netmgt.model.OnmsOutage;
+import org.opennms.netmgt.model.OnmsResource;
+import org.opennms.netmgt.model.OnmsResourceType;
+import org.opennms.netmgt.model.PrefabGraph;
+import org.opennms.netmgt.model.SurveillanceStatus;
+import org.opennms.web.api.Util;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.orm.ObjectRetrievalFailureException;
+import org.springframework.transaction.TransactionStatus;
+import org.springframework.transaction.support.TransactionCallback;
+import org.springframework.transaction.support.TransactionOperations;
+
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Collection;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.concurrent.Executors;
+
+/**
+ * Service class that encapsulate helper methods for surveillance views.
+ *
+ * @author Christian Pape
+ */
+public class DefaultSurveillanceViewService implements SurveillanceViewService {
+    /**
+     * The logger to be used
+     */
+    private static final Logger LOG = LoggerFactory.getLogger(DefaultSurveillanceViewService.class);
+    /**
+     * DAO instances injected via blueprint.xml
+     */
+    private NodeDao m_nodeDao;
+    private ResourceDao m_resourceDao;
+    private GraphDao m_graphDao;
```

Comment:
I see that the class name is DefaultSurveillanceViewService, is this necessary to pull in the GraphDao object. Here's why, its probably a semantics in naming of the class rather than the need. The Dashboard does have resource graphs but does the surveillance portion of the Dashboard need to look up this information? At the creation of this comment I have not looked at the rest of the class yet. I am open to your input.